src/Helpers/SAMLHelper.php

```diff
@@ -88,17 +92,31 @@ public function redirect(RequestHandler $requestHandler = null, HTTPRequest $req
     }
 
     /**
-     * Checks if the string is a valid guid in the format of A98C5A1E-A742-4808-96FA-6F409E799937
-     * Case in-sensitive
+     * Validates a NameID.
      *
-     * @param  string $guid
+     * @param  string $nameID
+     * @param  string $nameIDFormat
      * @return bool
      */
-    public function validGuid($guid)
+    public function validateNameID($nameID, $nameIDFormat): bool
     {
-        if (preg_match('/^[A-F0-9]{8}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{12}?$/i', $guid)) {
+        // Validate: Must fit within maximum size of DBVarchar.
+        $nameIDMaxSize = singleton(Member::class)->dbObject('GUID')->getSize();
+        if (strlen($nameID) > $nameIDMaxSize) {
+            return false;
+        }
+
+        // OK IF: We don't wish to do any further validation.
+        if (!Config::inst()->get(SAMLConfiguration::class, 'validate_nameid')) {
             return true;
         }
+
+        // OK IF: One of the registered extensions returns true.
+        $validationResults = $this->extend('updateNameIDValidation', $nameID, $nameIDFormat);
+        if ($validationResults && is_array($validationResults)) {
```

Comment:
I'm not sure this is within convention. Usually it takes just one extension to object for the result to fail, not for just one to pass.

This way we are most cautious and we encourage implementors to be cautious too. There's `false` for hard fail, `null` for no opinion and `true` for pass. We should aim to have no `false` and at least one `true`.

Reply:
Good point. I rarely call extend since in non-module code extend is a little overkill for design.

I think I just snatched a code-snippet from SS somewhere when I wrote this for project code.

My opinion is still that there should be no GUID validation at all (maybe apart from the varchar length). If I ever need to revisit our SAML implementation in project I'll try and finish this PR to upstream our changes. For now I think I'm going to close this due to time. Sorry!